### PR TITLE
Full Javadoc

### PR DESCRIPTION
**Pull request by Zabuzard**
Added full Javadoc coverage to the project.

Part of #81 . ~~Still blocked by #86 (as its based on that branch).~~

### CODE REVIEW

**Review by marko-radosavljevic — APPROVED** [review]
I love it! :relaxed: :heart:

Rarely, anyone wants to improve the docs, so thanks. ^^

File: application/src/main/java/org/togetherjava/tjbot/Application.java [hunk 1]
```diff
@@ -29,13 +33,20 @@ public static void main(final String[] args) {
         String token = args[0];
         String databasePath = args[1];
 
+        // Start
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/Application.java:36]: I don't think the comment really adds anything here

File: application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java [hunk 2]
```diff
@@ -12,11 +12,40 @@
...
+ * !dbget hello
+ * // TJ-Bot: Saved message: Hello World!
+ * }
+ * </pre>
+ */
 public final class DatabaseListener extends ListenerAdapter {
+    /**
+     * Logger for this class.
+     */
     private static final Logger logger = LoggerFactory.getLogger(DatabaseListener.class);
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java:37]: I don't think that comment adds anything.

File: application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java [hunk 3]
```diff
@@ -12,11 +12,40 @@
...
 
+    /**
+     * The database instance used by this command.
+     */
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java:41]: I don't think that comment adds anything.

File: application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java [hunk 4]
```diff
@@ -12,11 +12,40 @@
...
     private final Database database;
 
+    /**
+     * Creates a new command listener, using the given database
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java:45]:
```suggestion
     * Creates a new command listener, using the given database.
```

File: application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java [hunk 5]
```diff
@@ -51,11 +90,13 @@ private void handlePutMessage(String message, MessageReceivedEvent event) {
         String key = data[1];
         String value = data[2];
 
+        // Save the value in the database
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java:93]: That's probably not necessary either here? If you feel it is really needed I'd rather extract the try block in a method named .e.g `savePair`.

File: application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java [hunk 6]
```diff
@@ -78,12 +129,12 @@ private void handleGetMessage(String message, MessageReceivedEvent event) {
         }
         String key = data[1];
 
+        // Retrieve the value from the database
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java:132]: Same as above.

File: application/src/main/java/org/togetherjava/tjbot/PingPongListener.java [hunk 7]
```diff
@@ -6,9 +6,32 @@
...
+ * {@code
+ * !ping
+ * // TJ-Bot: Pong!
+ * }
+ * </pre>
+ */
 public final class PingPongListener extends ListenerAdapter {
+    /**
+     * Logger for this class
+     */
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/PingPongListener.java:26]: Doesn't really add anything.

File: application/src/main/java/org/togetherjava/tjbot/db/Database.java [hunk 8]
```diff
@@ -17,41 +16,56 @@
...
+ * connections are handled automatically by the system.
+ * <p>
+ * Instances of this class are thread-safe and can be used to concurrently write to the database.
  */
-public class Database {
+public final class Database {
 
+    /**
+     * The DSL context for this database.
+     */
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/db/Database.java:30]: Doesn't really add anything.

File: application/src/main/java/org/togetherjava/tjbot/db/Database.java [hunk 9]
```diff
@@ -17,41 +16,56 @@
...
     private final DSLContext dslContext;
+    /**
+     * Lock used to implement thread-safety across this class. Any database modifying method should
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/db/Database.java:33]:
```suggestion
     * Lock used to implement thread-safety across this class. Any database modifying method must
```

File: application/src/main/java/org/togetherjava/tjbot/db/Database.java [hunk 10]
```diff
@@ -17,41 +16,56 @@
...
-     * @param jdbcUrl the url to the database
+     * @param jdbcUrl the url to the database in the format expected by JDBC
      * @throws SQLException if no connection could be established
      */
     public Database(String jdbcUrl) throws SQLException {
         SQLiteConfig sqliteConfig = new SQLiteConfig();
         sqliteConfig.enforceForeignKeys(true);
-        // In WAL mode only concurrent writes pose a problem so we synchronize on those
-        sqliteConfig.setJournalMode(JournalMode.WAL);
+        // In WAL mode only concurrent writes pose a problem, so we synchronize on those
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/db/Database.java:47]:
```suggestion
        // In WAL mode only concurrent writes pose a problem, so we synchronize those
```

File: application/src/main/java/org/togetherjava/tjbot/db/Database.java [hunk 11]
```diff
@@ -168,20 +186,33 @@ public void readTransaction(CheckedConsumer<DSLContext, DataAccessException> han
...
+            handler.accept(dsl);
             return null;
         });
     }
 
     /**
-     * @return the database dsl context
+     * Gets the DSL context for this database.
+     *
+     * @return the DSL context
```
Comment (I-Al-Istannen) [inline application/src/main/java/org/togetherjava/tjbot/db/Database.java:202]: I don't think this adds anything?

**Review by I-Al-Istannen — DISMISSED** [review]
I'd comment public API, but inline comments / on private methods should add value or be left out IMHO. Otherwise they just clutter things and might go out of sync with the code.

File: application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java (hunk 2 above)
Comment (illuminator3) [reply application/src/main/java/org/togetherjava/tjbot/DatabaseListener.java:37]:
```java
/**
 * This class represents a cat
 */
class Cat {
}
```

**Review by illuminator3 — DISMISSED** [review]

**Review by illuminator3 — APPROVED** [review]

**Pull request merged**